screens/AppNavigator/screens/Dex/DexAddLiquidity.tsx

```diff
@@ -0,0 +1,225 @@
+        />
+        {renderIcon()}
+        <Text style={tailwind('w-12 ml-4 text-gray-500')}>{props.symbol}</Text>
+      </View>
+      <View style={tailwind('w-full bg-white flex-row border-t border-gray-200 h-12 items-center')}>
+        <View style={tailwind('flex flex-row flex-1 ml-4')}>
+          <Text>Balance: </Text>
+          <Text style={tailwind('text-gray-500')}>{props.balance.toString()}</Text>
+        </View>
+        <TouchableOpacity
+          style={tailwind('flex w-12 mr-4')}
+          onPress={onMax}
+        >
+          <Text style={[PrimaryColorStyle.text]}>MAX</Text>
+        </TouchableOpacity>
+      </View>
+    </View>
+  )
+}
+
+function Summary (props: { pair: ExtPoolPairData, sharePercentage: BigNumber }): JSX.Element {
+  const { pair, sharePercentage } = props
+  const RenderRow = (lhs: string, rhs: string): JSX.Element => {
+    return (
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium')}>{lhs}</Text>
+        </View>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium text-right text-gray-500')}>{rhs}</Text>
+        </View>
+      </View>
+    )
+  }
+
+  return (
+    <View style={tailwind('flex-col w-full items-center mt-4')}>
+      <View style={tailwind('bg-white p-2 border-b border-gray-200 flex-row items-center w-full h-16')}>
+        <View style={tailwind('flex-1')}>
+          <Text style={tailwind('font-medium')}>Price</Text>
+        </View>
+        <View style={tailwind('flex-1 flex-col')}>
+          <Text style={tailwind('font-medium text-right text-gray-600')}>{pair.aToBRate.toString()} {pair.aSymbol} per {pair.bSymbol}</Text>
+          <Text style={tailwind('font-medium text-right text-gray-600')}>{pair.bToARate.toString()} {pair.bSymbol} per {pair.aSymbol}</Text>
+        </View>
+      </View>
+      {RenderRow('Share of pool', sharePercentage.times(100).toString())}
+      {RenderRow(`Pooled ${pair.aSymbol}`, pair.tokenA.reserve)}
+      {RenderRow(`Pooled ${pair.bSymbol}`, pair.tokenB.reserve)}
+    </View>
+  )
+}
+
+function ContinueButton (props: { enabled: boolean, onPress: () => void }): JSX.Element {
+  const buttonColor = props.enabled ? PrimaryColorStyle.bg : { backgroundColor: 'gray' }
+  return (
+    <TouchableOpacity
+      style={[tailwind('m-2 p-3 rounded flex-row justify-center'), buttonColor]}
+      onPress={props.onPress}
+      disabled={!props.enabled}
+    >
+      <Text style={[tailwind('text-white font-bold')]}>Continue</Text>
+    </TouchableOpacity>
+  )
+}
+
+// TODO: display specific error
+function canAddLiquidity (pair: ExtPoolPairData, tokenAAmount: BigNumber, tokenBAmount: BigNumber, balanceA: BigNumber|undefined, balanceB: BigNumber|undefined): boolean {
+  if (tokenAAmount.isNaN() || tokenBAmount.isNaN()) {
```

Comment:
This can be improved with a `useForm` hook. We just need to add some validations, like required etc. Something like this -https://github.com/DeFiCh/wallet/blob/e29908f0f284a18aa9b2ca99fb8555a138d51203/screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx#L99

Reply:
I think probably not worth the effort to do this improvement now

Amount A and B will programmatically update each other, the value is actually processed before displayed. The processor `buildSummary()` callback must have both as dep list, else it will go infinite loop A -> B -> A -> ..... 
Adding the useForm controller here will need a lot of overhead code.

user input type is actually limited, required validation is quite minimal.